sktime/forecasting/base/adapters/_lagllama.py

```diff
@@ -0,0 +1,626 @@
+# todo: add an appropriate copyright notice for your estimator
+#       estimators contributed to sktime should have the copyright notice at the top
+#       estimators of your own do not need to have permissive or BSD-3 copyright
+
+__author__ = ["shlok191"]
+
+
+import subprocess
+
+from sktime.forecasting.base import BaseForecaster
+
+
+class _LagLlamaAdapter(BaseForecaster):
+    """Base class that interfaces the LagLlama forecaster.
+
+    Parameters
+    ----------
+    pretrained : bool, optional (default=True)
+        Dictates whether the pretrained LagLlama params should be initialized
+    """
+
+    _tags = {
+        "y_inner_mtype": "gluonts_PandasDataset_panel",
+        "X_inner_mtype": "gluonts_PandasDataset_panel",
+        "scitype:y": "both",
+        "ignores-exogeneous-X": True,
+        "requires-fh-in-fit": True,
+        "X-y-must-have-same-index": True,
+        "enforce_index_type": None,
+        "handles-missing-data": False,
+        "capability:insample": True,
+        "capability:pred_int": True,
+        "capability:pred_int:insample": True,
+        "authors": ["shlok191"],
+        "maintainers": ["shlok191"],
+        "python_version": None,
+        "python_dependencies": ["gluonts", "huggingface_hub", "lag_llama"],
+    }
+
+    def __init__(
+        self,
+        device=None,
+        context_length=None,
+        prediction_length=None,
+        num_samples=None,
+        batch_size=None,
+        nonnegative_pred_samples=None,
+        lr=None,
+        trainer_kwargs=None,
+    ):
+        import torch
+        from lag_llama.gluon.estimator import LagLlamaEstimator
+
+        # Defining private variable values
+        self.device = device
+        self.device_ = torch.device("cpu") if not device else device
+
+        self.context_length = context_length
+        self.context_length_ = 32 if not context_length else context_length
+
+        self.prediction_length = prediction_length
+        self.prediction_length_ = 100 if not prediction_length else prediction_length
+
+        self.num_samples = num_samples
+        self.num_samples_ = 10 if not num_samples else num_samples
+
+        self.batch_size = batch_size
+        self.batch_size_ = 32 if not batch_size else batch_size
+
+        # Now storing the training related variables
+        self.lr = lr
+        self.lr_ = 5e-5 if not lr else lr
+
+        self.trainer_kwargs = trainer_kwargs
+        self.trainer_kwargs_ = (
+            {"max_epochs": 50} if not trainer_kwargs else trainer_kwargs
+        )
+
+        # Not storing private variables for boolean specific values
+        self.nonnegative_pred_samples = nonnegative_pred_samples
+
+        super().__init__()
+
+        # Downloading the LagLlama weights from Hugging Face
+        download_command = (
+            "huggingface-cli download time-series-foundation"
+            + "-models/Lag-Llama lag-llama.ckpt --local-dir ."
+        )
+
+        status = subprocess.run(
```

Comment:
I assume there is also a way to download it from python. Please check this. If you do not find a way, please let me know, then I can check again.